ramips: Merge some duplicate cases in 02_network

adschm opened the pull request:
This does some quick tidy-up of the cases in 02_network.

Please have a look at the comments below.

adschm committed:
ramips: consolidate MAC address cases in 02_network

This just merges some duplicate definitions and consolidates lines.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adschm committed:
ramips: consistently use lower-case factory partition label

The vast majority of devices labels "factory" partition with lower
case. Convert the small fraction with capital letter to that and
merge another case in 02_network.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adrianschmutzler commented:
02_network contains some strange stuff:

https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/base-files/etc/board.d/02_network#L609
`lan_mac=$(cat /sys/class/net/eth0/address)`
Is this really set to itself or is there a deeper meaning to it that I do not understand.

https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/base-files/etc/board.d/02_network#L621
```
lan_mac=$(cat /sys/class/net/eth0/address)
lan_mac=$(macaddr_add "$lan_mac" 2)
```
Is this a workaround from times before mtd-mac-address-increment was available?

Similar here:
https://github.com/openwrt/openwrt/blob/master/target/linux/ramips/base-files/etc/board.d/02_network#L656
```
lan_mac=$(cat /sys/class/net/eth0/address)
lan_mac=$(macaddr_add "$lan_mac" -2)
```

What am I not getting here?

mkresin commented:
> What am I not getting here?

You miss that the mac address of the lan bridge is set here. If you bridge multiple interfaces with different mac addresses, it isn't that predictable which physical interface mac address will be used for the bridge. I know I wrote about my findings already, but right now I'm not able to find the mail.

Long story short, is it the mac address of the first added bridge member? The one of the last added member. The lowest mac address? The highest one? You get the point.

adrianschmutzler commented:
> Long story short, is it the mac address of the first added bridge member? The one of the last added member. The lowest mac address? The highest one? You get the point.

Thanks for the clarification.
I'm aware of that general problem, albeit those devices should only have one ethernet interface in LAN, so MAC address should be determined?

So this would only be required to KEEP the address in case the user adds interfaces later? This would however be different from the situation in ar71xx/ath79, where no lan MAC address is enforced for that case (if I get it correctly).

adrianschmutzler commented:
@mkresin 

> You miss that the mac address of the lan bridge is set here.

As pointed out in [1], I think "lan_mac" only set eth0.1, but not the bridge itself.
eth0.1 however should inherit from eth0?

[1]: https://www.mail-archive.com/openwrt-devel@lists.openwrt.org/msg47446.html

mkresin commented:
> As pointed out in 1, I think "lan_mac" only sets eth0.1, but not the bridge itself.

Indeed, I was wrong. /e/c/network from my HomeHub 5a has the following:

```
config device 'lan_dev'
	option name 'eth0.1'
	option macaddr 'ac:84:c9:a7:74:d4'
```

> eth0.1 however should inherit from eth0?

Yes, a vlan interface inherits the mac address from it's parent interface. Confirmed by a quick test.

> albeit those devices should only have one ethernet interface in LAN, so MAC address should be determined?

You forget about the wireless interface which will be a member of the lan bridge as well. It might be that the wireless mac address is used for the lan bridge.

In the end, it's even more complicated. Any traffic originating from (and to) the router itself, will have the mac address from the lan bridge set as src/dst mac address. Independent of which physical interface is used. Means, even packets transmitted over wireless will have the lan bridge mac address.

Of course, the wireless still broadcasts it's own mac-address in wireless beacons.

adrianschmutzler commented:
> Yes, a vlan interface inherits the mac address from it's parent interface. Confirmed by a quick test.

So, coming back to the initial issue, the following is indeed needless:
`lan_mac=$(cat /sys/class/net/eth0/address)`

adschm committed:
ramips: remove needless setting of lan_mac to eth0 in 02_network

This removes superfluous lines like
lan_mac=$(cat /sys/class/net/eth0/address)

Since lan_mac only sets the MAC address for eth0.1, these lines
can be safely removed as the address will be inherited from eth0
anyway.

Signed-off-by: Adrian Schmutzler <freifunk@adrianschmutzler.de>

adrianschmutzler commented:
Rebased and fixed commit title capitalization.

blocktrron commented on target/linux/ramips/base-files/etc/board.d/02_network:
```diff
@@ -606,24 +602,25 @@ ramips_setup_macs()
 	hootoo,ht-tm02|\
 	loewe,wmdr-143n|\
 	unbranded,a5-v11)
-		lan_mac=$(cat /sys/class/net/eth0/address)
```
This clause should be safe to delete entirely, shouldn't it?

adrianschmutzler replied:
@blocktrron 
you have to keep the case, as the default case will set up WAN MAC = LAN_MAC + 1
But you can obviously delete the line setting lan_mac.
That's why I made it an empty case in my patch 3 here.

I personally do not consider the default case very helpful here, since it will make many device support patch authors fall into it by just overlooking (similar to the discussion with the ralink_* size variables). However, I could think of no way to change this without having to check every single device in the target or by introducing a huge non-default case for 150 to 200 devices.

blocktrron replied:
All 3 devices only have 1 ethernet port, thus setting the WAN mac is kind of pointless.

981213 replied:
oh... and I made a patch dropping it...

adrianschmutzler replied:
Well, it will be a dead additional entry in uci config then, if the config_generate script isn't even smart enough to not produce it then. So it shouldn't break anything.
Still, IMHO keeping the empty case would be the proper solution. Other one-port devices now falling into default case by accident or newly added might use it in the future, too.

adrianschmutzler replied:
And the really proper solution would be to remove the default-case eventually ...

981213 commented:
Just noticed that I introduced another duplication for Phicomm K2P (didn't notice that 0xe000=57344 and 0xe006=57350). Since I'll fix it anyway I decided to also pick two commits here :P
I'm not quite sure if I should merge the third one.

981213 commented:
More problems about that one:
```
       hootoo,ht-tm02|\
       loewe,wmdr-143n|\
       unbranded,a5-v11)
               ;;
```
All these three devices have one eth port only. If we need this case we might have to add the other routers (actually a lot) with one port into this case.

adrianschmutzler commented:
@981213 
Sorry for repeating that theme, but this might be good reason to really invert logic in that case:
The current default case would then be explicit and all the one-port devices would fall into a (non-existant) default case without setup.
I'm just still looking for a solution to automate the identification of one-port devices here, so I would not have to check individually.

blocktrron commented:
@981213 If the device has no WAN interface defined in `ramips_setup_interfaces`, we should be safe dropping this case. We define a WAN MAC in the board.json, but theres no WAN interface defined, we have no issue there.
